pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -1064,18 +1079,16 @@ impl Timeline {
                 } else if *lsn <= horizon {
                     match val {
                         crate::repository::Value::Image(image) => {
-                            if lsn <= &horizon {
-                                base_image = Some((*lsn, image.clone()));
-                                break;
-                            }
+                            base_image = Some((*lsn, image.clone()));
+                            break;
                         }
                         crate::repository::Value::WalRecord(wal) => {
                             delta_above_base_image.push((*lsn, wal.clone()));
                         }
                     }
                 }
             }
-            delta_above_base_image.reverse();
```

Comment:
This change seems like a correctness fix, orthogonal to what this PR is about.

I guess you found this because of the `test_gc_feedback`?
Is there a corresponding unit test?

Reply:
Yes. It's tested as part of the benchmark, but I can also add a unit test for that based on NeonWalRecord.

Reply:
Given that it's so essential, let's have that one. Can be done in a follow-up.

Reply:
will do as a follow up